app/models/application_record.rb

```diff
@@ -137,4 +147,29 @@ def self.ethnicity_eeo_data(records)
       [ethnicity, specific_records.count, specific_records.interview_count]
     end
   end
+
+  def self.notification_emails(records)
+    emails = Hash.new { |h, k| h[k] = {} }
+    records.each do |record|
+      emails = build_email_hash(emails, record)
+    end
+    send_emails(emails)
+  end
+
+  def self.send_emails(emails)
```

Comment:
The word `email` is way too prevalent here. You could help fix this by iterating with `each_pair` and then assigning a meaningful name to the value, since you do call `emails[email]` in the loop.